lms/djangoapps/courseware/views/views.py

```diff
@@ -863,6 +864,126 @@ def post(self, request, course_id):
+                    course_organization = course_overview.display_org_with_default
+
+                modes = CourseMode.modes_for_course_dict(course_key)
+                registered = registered_for_course(course_overview, request.user)
+
+                if request.user.has_perm(VIEW_COURSE_HOME, course_overview):
+                    course_target = reverse(course_home_url_name(course_overview.id), args=[text_type(course_overview.id)])
+                else:
+                    course_target = reverse('about_course', args=[text_type(course_overview.id)])
+
+                show_courseware_link = bool(
+                    (
+                        request.user.has_perm(VIEW_COURSEWARE, course_overview)
+                    ) or settings.FEATURES.get('ENABLE_LMS_MIGRATION')
+                )
+
+                can_enroll = bool(request.user.has_perm(ENROLL_IN_COURSE, course_overview))
+                invitation_only = course_overview.invitation_only
+                is_course_full = CourseEnrollment.objects.is_course_full(course_overview)
+                active_reg_button = not (registered or is_course_full or not can_enroll)
+                ecomm_service = EcommerceService()
+                ecommerce_checkout = ecomm_service.is_enabled(request.user)
+                ecommerce_checkout_link = ''
+                ecommerce_bulk_checkout_link = ''
+                single_paid_mode = None
+                if ecommerce_checkout:
+                    if len(modes) == 1 and list(modes.values())[0].min_price:
+                        single_paid_mode = list(modes.values())[0]
+                    else:
+                        # have professional ignore other modes for historical reasons
+                        single_paid_mode = modes.get(CourseMode.PROFESSIONAL)
+
+                    if single_paid_mode and single_paid_mode.sku:
+                        ecommerce_checkout_link = ecomm_service.get_checkout_page_url(single_paid_mode.sku)
+                    if single_paid_mode and single_paid_mode.bulk_sku:
+                        ecommerce_bulk_checkout_link = ecomm_service.get_checkout_page_url(single_paid_mode.bulk_sku)
+
+                allow_anonymous = check_public_access(course_overview, [COURSE_VISIBILITY_PUBLIC, COURSE_VISIBILITY_PUBLIC_OUTLINE])
+                is_shib_course = uses_shib(course_overview)
+
+                # add staff members of each course run in staff list
+                for staff in course_run.get('staff'):
+                    uuid = staff.get('uuid')
+                    if uuid not in staff_ids:
+                        course_staff.append({
+                            'name': f"{staff.get('given_name')} {staff.get('family_name')}",
+                            'bio': staff.get('bio'),
+                            'profile_image_url': staff.get('profile_image_url')
+                        })
+                        staff_ids.add(uuid)
+
+                course_runs_context.append({
```

Comment:
can we simply assign this `dict` to `course_runs_context`? because until this point `course_runs_context` is an empty list

Reply:
there can be multiple runs so dict for each run will be appended to this list